src/expand.cpp

```diff
@@ -86,6 +87,7 @@ namespace Sass {
 
   Statement* Expand::operator()(Ruleset* r)
   {
+    bool old_at_root_without_rule = this->at_root_without_rule;
```

Comment:
You should be able to write that as `LOCAL_FLAG(at_root_without_rule, false);`
The macro will create a small stack object that will ensure the flag is reset once it goes out of scope. IMO that is much safer than resetting the flag on your own, since it will also handle cases where we exit via a throw. I have added that [Macro (and the helper class)] [1] some time ago.

[1]: https://github.com/sass/libsass/blob/826827a6a52b107d471958fc34dcafbb847ea6e4/src/ast_def_macros.hpp#L5-L27